recmetrics/metrics.py

```diff
@@ -139,7 +139,69 @@ def _ark(actual: list, predicted: list, k=10) -> int:
 
     return score / len(actual)
 
-def mark(actual: List[list], predicted: List[list], k=10) -> int:
+def _pk(actual: list, predicted: list, k) -> float:
+    """
+    Computes precision at k.
+    Parameters
+    ----------
+    actual : list
+        A list of actual items to be predicted
+    predicted : list
+        An ordered list of predicted items
+    k : int
+        Number of predictions to consider
+    Returns:
+    -------
+    score : float
+        The precision at k.
+    """
+
+    if len(predicted) > k:
+        predicted = predicted[:k]
+
+    if not predicted or not actual:
+        return 0.0
+        
+    return sum([item in actual for item in set(predicted)]) / len(predicted)   
```

Comment:
Note: I think that sometimes people divide by `k`, or sometimes divide by `min(k, len(predicted))`. AFAICT, this is an important implementation decision?